code/modules/projectiles/guns/projectile/rocketlauncher.dm

```diff
@@ -22,16 +23,51 @@
 	attack_verb = list("strikes", "hits", "bashes")
 	gun_flags = 0
 
+/obj/item/weapon/gun/projectile/rocketlauncher/New()
+	..()
+	update_icon()
+
 /obj/item/weapon/gun/projectile/rocketlauncher/isHandgun()
 	return FALSE
 
 /obj/item/weapon/gun/projectile/rocketlauncher/update_icon()
-	if(!getAmmo())
-		icon_state = "rpg_e"
-		item_state = "rpg_e"
-	else
-		icon_state = "rpg"
-		item_state = "rpg"
+	if(!getAmmo()) //empty
+		overlays.len = 0 //remove missile overlays
+		item_state = initial_icon
+		item_state = "[item_state]_e"
+	else //not empty
+		overlays.len = 0
+		var/obj/item/ammo_casing/projectile = loaded[1]
```

Comment:
```suggestion
		var/obj/item/ammo_casing/rocket_rpg/projectile = loaded[1]
```

Reply:
yeah I literally typed it out in discord correctly then went and mistyped it in the code.